### PR TITLE
blob/fileblob: improve path munging on Windows, including tests/examples

### PR DESCRIPTION
**Pull request by vangent**

### CODE REVIEW

**Comment by vangent** [comment]
(These tests pass on Windows).

File: blob/fileblob/example_test.go [hunk 1]
```diff
@@ -65,9 +66,16 @@ func Example_openBucket() {
...
+	// On Windows, convert "\" to "/" and add a leading "/":
+	dirpath := filepath.ToSlash(dir)
+	if os.PathSeparator != '/' && !strings.HasPrefix(dirpath, "/") {
+		dirpath = "/" + dirpath
+	}
+
 	// OpenBucket creates a *blob.Bucket from a URL.
 	ctx := context.Background()
-	b, err := blob.OpenBucket(ctx, path.Join("file://", dir))
+	b, err := blob.OpenBucket(ctx, "file://"+dirpath)
```
Comment (vangent) [inline blob/fileblob/example_test.go]: `path.Join` is the wrong thing here, as it collapses multiple slashes, resulting in URLs like `file:/tmp/dir`.

**Comment by vangent** [comment]
@zombiezen moving review to @clausti since I've sent a lot your way recently. Feel free to take a look if you'd like.

File: blob/fileblob/example_test.go [hunk 2]
```diff
@@ -65,9 +66,16 @@ func Example_openBucket() {
 	}
 	defer os.RemoveAll(dir)
 
+	// On Unix, append the dir to "file://".
+	// On Windows, convert "\" to "/" and add a leading "/":
+	dirpath := filepath.ToSlash(dir)
```
Comment (zombiezen) [inline blob/fileblob/example_test.go]: Optional: Seeing this in the example directly, it kinda feels like we should have a utility function that does this, although I realize the user would probably just call the `fileblob.OpenBucket` call directly. WDYT?

File: blob/fileblob/example_test.go (hunk 1 above)
Comment (zombiezen) [reply blob/fileblob/example_test.go]: Ack.

File: blob/fileblob/example_test.go (hunk 2 above)
Comment (vangent) [reply blob/fileblob/example_test.go]: I think the user is more likely to be constructing the URLs by hand than to be writing code to generate them. I agree with you that if they have the directory in code, they should just call `OpenBucket` directly, and not go through the URL path.

**Comment by vangent** [comment]
gentle ping @clausti ?

**Pull request merged**